### PR TITLE
Fix to Diagnosis search and Add Age Range Filters

### PR DESCRIPTION
**Pull request by growlingflea**
Updates to two of the new reports for NHanes and general use.

### CODE REVIEW

File: interface/reports/username_report.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * interface/reports/username_report.php - User Activity from Log Table
+ * interface/reports/username_report.php List number of documents.
```
Comment (tmccormi) [inline interface/reports/username_report.php]: This is a reversion of changes that were already merged

File: library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
 /**
- * library/ajax/clinical_stats_and_lab_stats_by_demographics.php: handles ajax calls from demographic lab/procedure reports
- * file adapted to present user activity log
+ * library/ajax/clinical_stats_and_lab_stats_by_demographics.php: handles ajax calls from demographic lab and procedure reports
+ *
```
Comment (tmccormi) [inline library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php]: This is a reversion of changes that were already merged

File: library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php [hunk 3]
```diff
@@ -57,23 +57,46 @@ function ifTestingTrue($testing){
...
-    $query .= "join form_encounter on issue_encounter.encounter = form_encounter.encounter";
+     if(! empty($_POST['diag'])){
+       array_push($queryArray, $_POST['diag'] );
+       $query .= ' AND diagnosis = ?';
+     }
 
 
-    ini_set('memory_limit', '1000M');
-    $result = sqlStatement($query);
+    if(! empty($_POST['ethnicity'])){
```
Comment (tmccormi) [inline library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php]: Is the limit overide not needed any more?

File: interface/reports/clinical_stats_by_demographics_report.php [hunk 4]
```diff
@@ -155,9 +125,14 @@ function show_all_diags(){
             type: "POST",
             url: "../../library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php",
             data: {
+
                 func:"get_all_diags_data",
-                diag:"<?php  echo $_POST['form_diagnosis']   ?>",
-                proc:"<?php  echo $_POST['form_service_codes']   ?>"
+                diag:"<?php  echo $_POST['form_diagnosis'];   ?>",
+                proc:"<?php  echo $_POST['form_service_codes'];   ?>",
```
Comment (tmccormi) [inline interface/reports/clinical_stats_by_demographics_report.php]: Why is this field in this report, it's not used here

Comment (tmccormi) [reply interface/reports/clinical_stats_by_demographics_report.php]: specifically "form_service_codes"

Comment (growlingflea) [reply interface/reports/clinical_stats_by_demographics_report.php]: Removed it

File: library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php (hunk 2 above)
Comment (growlingflea) [reply library/ajax/clinical_stats_and_lab_stats_by_demographics_report_ajax.php]: fixed

**Comment by tmccormi** [comment]
@aethelwulffe or @teryhill please review or let me know if it's OK to push these enhancements for Dr Hoyt

**Comment by tmccormi** [comment]
Need to merge this... will do that tomorrow if no one objects

**Pull request merged**